src/elements/content-sidebar/versions/VersionsSidebarContainer.js

```diff
@@ -163,9 +166,12 @@ class VersionsSidebarContainer extends React.Component<Props, State> {
         this.mergeResponse(data);
     };
 
-    handleFetchError = (): void => {
+    handleFetchError = (error: Object): void => {
+        const { showVersionErrorWithUpsell } = this.props;
+        const shouldShowVersionErrorWithUpsell = !!showVersionErrorWithUpsell && error.status === 403;
         this.setState({
-            error: messages.versionFetchError,
+            error: shouldShowVersionErrorWithUpsell ? messages.versionNotAvailable : messages.versionFetchError,
```

Comment:
Do we need this boolean or can we key off the existence of `onUpgradeClick`, similar to the underlying components?

Reply:
I thought it was but I thought through the three different scenarios where the onUpgradeClick will not be null and with checking for the error code, and the onUpgradeClick existing we can in fact infer the correct bucket and omit that prop!